pkg/node/store/store.go

```diff
@@ -102,18 +144,61 @@ func (nr *NodeRegistrar) RegisterNode(n *nodeTypes.Node, manager NodeManager) er
 		return err
 	}
 
-	if err = store.UpdateLocalKeySync(context.TODO(), n); err != nil {
-		store.Release()
+	var registerStore *store.SharedStore
+	if option.Config.JoinK8sNamespace != "" {
+		nodeId := make(chan uint32, 1)
+
+		// Join the shared store for node registrations
+		registerStore, err := store.JoinSharedStore(store.Configuration{
+			Prefix:     NodeRegisterStorePrefix,
+			KeyCreator: KeyCreator,
+			Observer:   NewRegisterObserver(n.Fullname(), nodeId),
+		})
+		if err != nil {
+			nodeStore.Release()
+			return err
+		}
+
+		// drain the channel of old updates first
```

Comment:
Which old updates? We have just registered the `SharedStore`

Reply:
the registration of `SharedStore` will dump the keys from the KV store before returning, which will return any old NodeIdentity if it exists. Since the labels on this run can be different (hence leading to a different numeric identity) we must disregard any such identities received before the new registration.